torchaudio/datasets/tedlium.py

```diff
@@ -0,0 +1,133 @@
+        "url": "http://www.openslr.org/resources/51/TEDLIUM_release-3.tgz",
+        "checksum": "685d27c39c53217383d7933cc405a07048004127",
+        "data_path": "data/",
+        "subset": None,
+    },
+}
+
+
+def load_tedlium_item(
+    fileid: str, line: int, path: str, ext_audio: str, ext_txt: str
+) -> Tuple[Tensor, int, str, int, int, int]:
+    transcript_path = os.path.join(path, "stm/", fileid)
+    with open(transcript_path + ext_txt) as f:
+        transcript = f.readlines()[line]
+        talk_id, _, speaker_id, start_time, end_time, identifier, transcript = transcript.split(
+            " ", 6
+        )
+
+    wave_path = os.path.join(path, "sph/", fileid)
+    waveform, sample_rate = torchaudio.load(wave_path + ext_audio)
+    print(wave_path + ext_audio)
+    # Calculate indexes for start time and endtime
+    start_time = int(float(start_time) * sample_rate)
+    end_time = int(float(end_time) * sample_rate)
+    print(start_time, end_time)
+    waveform = waveform[:, start_time:end_time]
+    return (
+        waveform,
+        sample_rate,
+        transcript,
+        talk_id,
+        speaker_id,
+        identifier,
+        transcript,
+    )
+
+
+class TEDLIUM(Dataset):
+    """
+    Create a Dataset for Tedlium. Each item is a tuple of the form:
+    waveform, sample_rate, utterance, speaker_id, chapter_id, utterance_id
+    """
+
+    _ext_txt = ".stm"
+    _ext_audio = ".sph"
+    _folder_audio = "sph/"
+    _folder_txt = "stm/"
+
+    def __init__(
+        self,
+        root: str,
+        release: str = RELEASE,
+        subset: str = None,
+        folder_in_archive: str = _RELEASE_CONFIGS[RELEASE]["folder_in_archive"],
+        download: bool = False,
+    ) -> None:
+
+        if release in _RELEASE_CONFIGS.keys():
+            folder_in_archive = _RELEASE_CONFIGS[release]["folder_in_archive"]
+            url = _RELEASE_CONFIGS[release]["url"]
+            subset = subset if subset else _RELEASE_CONFIGS[release]["subset"]
+        else:
+            # Raise warning
+            raise RuntimeError(
+                "The release {} does not match any of the supported tedlium releases{} ".format(
+                    filepath, _RELEASE_CONFIGS.keys(),
+                )
+            )
+
+        basename = os.path.basename(url)
+        archive = os.path.join(root, basename)
+
+        basename = basename.split(".")[0]
+
+        self._path = os.path.join(root, folder_in_archive, _RELEASE_CONFIGS[release]["data_path"])
+        if subset in ["train", "dev", "test"]:
+            self._path = os.path.join(self._path, subset + "/")
+        if download:
+            if not os.path.isdir(self._path):
+                if not os.path.isfile(archive):
+                    checksum = _CHECKSUMS.get(url, None)
+                    download_url(url, root, hash_value=checksum)
+                extract_archive(archive)
+
+        walker = walk_files(self._path, suffix=self._ext_txt, prefix=False, remove_suffix=True)
```

Comment:
I am downloading all TEDLIUM datasets now so that we can discuss the detail of how to traverse these directories. I will give my thoughts once I see the dataset strucutres.

Reply:
That would be perfect, lets have a discussion for it when you have the data.